diff --git a/klippy/extras/display/menu.py b/klippy/extras/display/menu.py
index e7723a7e3..261990761 100644
--- a/klippy/extras/display/menu.py
+++ b/klippy/extras/display/menu.py
@@ -1026,6 +1026,8 @@ class MenuManager:
             self.begin(eventtime)
 
     def key_event(self, key, eventtime):
+        self.printer.send_event('menu:%s' % (key), key, eventtime) # granular
+        self.printer.send_event('menu:action', key, eventtime) # any item
         if key == 'click':
             self._click_callback(eventtime, key)
         elif key == 'long_click':
